### PR TITLE
small flakyness fixes

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why this change is needed

our tests have become very flaky

### What changes were made as part of this PR

Lots of small changes. Please see the messages in the file

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/evm/ethchainadapter/eth_chainadapter.go [hunk 1]
```diff
@@ -56,15 +56,12 @@ func (e *EthChainAdapter) CurrentBlock() *gethtypes.Header {
 	if currentBatchSeqNo == nil {
 		return nil
 	}
-	ctx, cancelCtx := context.WithTimeout(context.Background(), e.config.RPCTimeout)
-	defer cancelCtx()
-
-	currentBatch, err := e.storage.FetchBatchHeaderBySeqNo(ctx, currentBatchSeqNo.Uint64())
+	currentBatch, err := e.storage.FetchBatchHeaderBySeqNo(context.Background(), currentBatchSeqNo.Uint64())
```
Comment (tudor-malene) [inline go/enclave/evm/ethchainadapter/eth_chainadapter.go:59]: given this function doesn't return an error, it's safest to not use timeouts.

I've seen it fail with "context cancelled" which ended up in a panic

File: go/enclave/txpool/txpool.go [hunk 2]
```diff
@@ -70,7 +70,11 @@ func (t *TxPool) Start() error {
 // PendingTransactions returns all pending transactions grouped per address and ordered per nonce
 func (t *TxPool) PendingTransactions() map[gethcommon.Address][]*gethtxpool.LazyTransaction {
 	// todo - for now using the base fee from the block
-	baseFee := t.Chain.CurrentBlock().BaseFee
+	currentBlock := t.Chain.CurrentBlock()
+	if currentBlock == nil {
```
Comment (tudor-malene) [inline go/enclave/txpool/txpool.go:74]: the currentBlock can be nil.

File: integration/ethereummock/gethutil.go [hunk 3]
```diff
@@ -18,6 +18,9 @@ var EmptyHash = gethcommon.Hash{}
 // it also returns the blocks that became canonical, and the once that are now the fork
 func LCA(ctx context.Context, newCanonical *types.Block, oldCanonical *types.Block, resolver *blockResolverInMem) (*common.ChainFork, error) {
 	b, cp, ncp, err := internalLCA(ctx, newCanonical, oldCanonical, resolver, []common.L1BlockHash{}, []common.L1BlockHash{})
+	if err != nil {
```
Comment (tudor-malene) [inline integration/ethereummock/gethutil.go:21]: we didn't handle this error, which cause NPE

File: integration/ethereummock/node.go [hunk 4]
```diff
@@ -366,14 +368,15 @@ func (m *Node) processBlock(b *types.Block, head *types.Block) *types.Block {
 		m.stats.L1Reorg(m.l2ID)
 		fork, err := LCA(context.Background(), head, b, m.BlockResolver)
 		if err != nil {
-			panic(err)
+			m.logger.Error("Should not happen.", log.ErrKey, err)
```
Comment (tudor-malene) [inline integration/ethereummock/node.go:371]: seems to recover without the panic

File: go/enclave/enclave.go [hunk 5]
```diff
@@ -116,7 +116,7 @@ func NewEnclave(
 	}
 
 	// Initialise the database
-	cachingService := storage.NewCacheService(logger)
+	cachingService := storage.NewCacheService(logger, config.UseInMemoryDB)
```
Comment (tudor-malene) [inline go/enclave/enclave.go:119]: reduce the memory footprint of the cache

File: go/enclave/evm/ethchainadapter/eth_chainadapter.go [hunk 6]
```diff
@@ -56,15 +56,12 @@ func (e *EthChainAdapter) CurrentBlock() *gethtypes.Header {
...
 	if err != nil {
 		e.logger.Warn("unable to retrieve batch seq no", "currentBatchSeqNo", currentBatchSeqNo, log.ErrKey, err)
 		return nil
 	}
-	batch, err := e.gethEncoding.CreateEthHeaderForBatch(ctx, currentBatch)
+	batch, err := e.gethEncoding.CreateEthHeaderForBatch(context.Background(), currentBatch)
```
Comment (tudor-malene) [inline go/enclave/evm/ethchainadapter/eth_chainadapter.go:64]: avoid the timeout

File: go/enclave/storage/enclavedb/block.go [hunk 7]
```diff
@@ -56,8 +56,8 @@ func IsCanonicalBlock(ctx context.Context, dbtx *sql.Tx, hash *gethcommon.Hash)
 }
 
 // CheckCanonicalValidity - expensive but useful for debugging races
-func CheckCanonicalValidity(ctx context.Context, dbtx *sql.Tx) error {
-	rows, err := dbtx.QueryContext(ctx, "select count(*), height from batch where is_canonical=true group by height having count(*) >1")
+func CheckCanonicalValidity(ctx context.Context, dbtx *sql.Tx, blockId int64) error {
```
Comment (tudor-malene) [inline go/enclave/storage/enclavedb/block.go:59]: we have a bug where we end up with non-canonical batches that are marked as canonical, which breaks the rollup compression/decompression and shows up as some weird xchain panic.

This should help narrow it down

File: go/host/enclave/guardian.go [hunk 8]
```diff
@@ -558,7 +558,7 @@ func (g *Guardian) periodicBatchProduction() {
 			skipBatchIfEmpty := g.maxBatchInterval > g.batchInterval && time.Since(g.lastBatchCreated) < g.maxBatchInterval
 			err := g.enclaveClient.CreateBatch(context.Background(), skipBatchIfEmpty)
 			if err != nil {
-				g.logger.Error("Unable to produce batch", log.ErrKey, err)
+				g.logger.Crit("Unable to produce batch", log.ErrKey, err)
```
Comment (tudor-malene) [inline go/host/enclave/guardian.go:561]: this is catastrophic, so it's good to know early

File: integration/smartcontract/debug_wallet.go [hunk 9]
```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/ten-protocol/go-ten/go/wallet"
 )
 
-var _timeout = 120 * time.Second
+var _timeout = 180 * time.Second
```
Comment (tudor-malene) [inline integration/smartcontract/debug_wallet.go:15]: hopefully avoid a timeout

File: integration/ethereummock/gethutil.go (hunk 3 above)
Comment (badgersrus) [reply integration/ethereummock/gethutil.go:21]: good spot!

**Review by badgersrus — APPROVED** [review]
lgtm, some really good spots in here!

**Pull request merged**